### PR TITLE
feat(clock): small analogue clock face

### PR DESCRIPTION
**Pull request by bualoy-napat**
## Description
A new feature on the clock. When the width is less than 130px, the face of the clock will change to a smaller face that is easy to read.

Update UI of digital and analog clock to match with current spec from UX.

## Type of change
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: packages/elements/src/clock/index.ts [hunk 1]
```diff
@@ -557,6 +569,17 @@ export class Clock extends BasicElement {
...
 
+  /**
+   * Called when the element's dimension have changed
+   * @param size Element size
+   * @returns {void}
+   */
+  public resizedCallback (size: ElementSize): void {
+    this.size = size.width < 130;
+    this.width = size.width;
+    this.requestUpdate();
```
[the remark this answers is not in the document]
Comment (dtanp-rft) [reply packages/elements/src/clock/index.ts:580]: Ah that's right! I think this is a good solution as we have discussed that `size` can be (hidden) prop. We should try to go with this approach.

[the remark this answers is not in the document]
Comment (bualoy-napat) [reply packages/elements/src/clock/index.ts:580]: Thank you for your suggestion. It's very helpful.

File: packages/elements/src/clock/index.ts [hunk 2]
```diff
@@ -611,10 +634,19 @@ export class Clock extends BasicElement {
...
-
+    this.size ? this.setAttribute('size', 'small') : this.removeAttribute('size');
+    
+    if(this.size) {
+      this.style.fontSize = (this.width / 2).toString() + 'px'; // Calculate font-size for small size of the clock. We cannot set font-size rely on the width of element (`vw` unit only work on the width of the browser).
+    }
+    else {
+      this.style.fontSize = '2.5em'; // Need to set font-size to default if the size is not small.
+    }
+
```
[the remark this answers is not in the document]
Comment (bualoy-napat) [reply packages/elements/src/clock/index.ts:632]: Already fix in recently commit.

File: packages/elements/src/clock/__test__/clock.analogue.test.js [hunk 3]
```diff
@@ -44,5 +44,12 @@ describe('clock/Analogue', () => {
 
       expect(getClockHand('second').style.transform, 'seconds hand should have 270 degrees angle').to.be.equal('rotate(270deg)');
     });
+    it('Shows small size clock when width is less than 130px', async () => {
+      el.style.width = '129px';
+      await elementUpdated(el);
+      await nextFrame();
+
+      expect(el.hasAttribute('size')).to.equal(true, 'attribute "size" should be present');
```
[the remark this answers is not in the document]
Comment (bualoy-napat) [reply packages/elements/src/clock/__test__/clock.analogue.test.js]: I have already improved on this point with a recent commit.

**Review by dtanp-rft — CHANGES_REQUESTED** [review]
I think `size=small` should not be added if it's not analog clock.
Would you please change it and write test for this change?

Regarding to the test, I suggest you breakdown test case to be smaller test. When test is failed, it's easier to understand what is going wrong e.g.

**Shows small size clock when width is less than 130px**
1. test if [part=digital] is rendered if width >= 130px
2. set width < 130px
3. test if [part=digital] is not rendered
4. test if [part="segment am-pm"] is rendered (ps. I wouldn't bother to check if it needs to be inside [part="hands"] but it's fine if you want the test is a little bit strict

**Small size clock show AM/PM if it has attribute `am-pm`**
1. Use el = await fixture('<ef-clock am-pm analogue></ef-clock>');
2. Set width <= 130px 
3. Test if am-pm part is rendered

**Attribute `size=small` shouldn't show if it's not analog clock**
1. el = await fixture('<ef-clock></ef-clock>');
2. Set width < 130px
3. Test if `size=small` is not present
4. Set width >= 130px
5. Test if `size=small` is not present

**Review by dtanp-rft — APPROVED** [review]
Please squash commit when merge.

**Pull request merged**